src/base/Context.ts

```diff
@@ -629,32 +629,31 @@ export class Context {
 
     const claimData = await P.map(claim1stKeys, async claim1stKey => {
       const entries = await identity.claims.entries(claim1stKey);
-      const compactData = compact(
-        entries.map(
-          ([
-            key,
-            { claim_issuer: claimissuer, issuance_date: issuanceDate, expiry: rawExpiry, claim },
-          ]) => {
-            const { target } = key.args[0] as Claim1stKey;
-            const expiry = rawExpiry ? momentToDate(rawExpiry.unwrap()) : null;
-            if ((!includeExpired && (expiry === null || expiry > new Date())) || includeExpired) {
-              return {
-                target: new Identity({ did: identityIdToString(target) }, this),
-                issuer: new Identity({ did: identityIdToString(claimissuer) }, this),
-                issuedAt: momentToDate(issuanceDate),
-                expiry,
-                claim: meshClaimToClaim(claim),
-              };
-            }
-            return undefined;
+      const data: (ClaimData | undefined)[] = [];
+      entries.forEach(
+        ([
+          key,
+          { claim_issuer: claimissuer, issuance_date: issuanceDate, expiry: rawExpiry, claim },
+        ]) => {
+          const { target } = key.args[0] as Claim1stKey;
+          const expiry = rawExpiry ? momentToDate(rawExpiry.unwrap()) : null;
+          if ((!includeExpired && (expiry === null || expiry > new Date())) || includeExpired) {
+            data.push({
+              target: new Identity({ did: identityIdToString(target) }, this),
+              issuer: new Identity({ did: identityIdToString(claimissuer) }, this),
+              issuedAt: momentToDate(issuanceDate),
+              expiry,
+              claim: meshClaimToClaim(claim),
+            });
+          } else {
+            data.push(undefined);
           }
-        )
+        }
       );
-
-      return compactData;
+      return compact(data);
```

Comment:
Since you're using `forEach`, you no longer need the `undefined` values nor `compact`